cpp/include/cudf/io/parquet.hpp

```diff
@@ -821,13 +760,7 @@ class parquet_writer_options_builder {
    * be same size as number of sinks in sink_info
    * @return this for chaining
    */
-  parquet_writer_options_builder& partitions(std::vector<partition_info> partitions)
-  {
-    CUDF_EXPECTS(partitions.size() == options._sink.num_sinks(),
-                 "Mismatch between number of sinks and number of partitions");
-    options.set_partitions(std::move(partitions));
-    return *this;
-  }
+  parquet_writer_options_builder& partitions(std::vector<partition_info> partitions);
```

Comment:
The builder function is duplicating validation in the `options` setter...would it be better to just remove the `CUDF_EXPECTS` here (and in `key_value_metadata` and `column_chunks_file_paths`)?

Reply:
Yes, that makes sense.